lib/internal/test_runner/runner.js

```diff
@@ -425,6 +433,22 @@ function run(options) {
   if (watch != null) {
     validateBoolean(watch, 'options.watch');
   }
+  if (shard != null) {
+    validateObject(shard, 'options.shard');
+    // Avoid re-evaluating the shard object in case it's a getter
+    shard = { __proto__: null, index: shard.index, total: shard.total };
+
+    validateInteger(shard.total, 'options.shard.total', 1);
+    validateInteger(shard.index, 'options.shard.index');
+
+    if (shard.index <= 0 || shard.total < shard.index) {
+      throw new ERR_OUT_OF_RANGE('options.shard.index', `>= 1 && <= ${shard.total} ("options.shard.total")`, shard.index);
+    }
+
+    if (watch) {
```

Comment:
q: why?

Reply:
why would you use sharding if you need to watch all the files? watch is for active development

Reply:
You had a bug in a certain shard in CI and you want to iteratively fix it? Is there a reason not to allow it?

Reply:
if you add a file it will change the sharding, we can add it I just think it won't be really useful